### DISCUSSION TITLE
Easiest way to get the wallet provider name (e.g. MetaMask, Trust Wallet, Rainbow, etc...) using Web3Modal

### DISCUSSION THREAD

**Discussion by ghost** [WalletConnect/walletconnect-monorepo#3763] [Developer Support]
Currently, we are using the wagmi useAccount hook which returns a connector. The connector has a name property, which in the case of MetaMask, correctly returns 'MetaMask'. However in the case of 'Trust Wallet', the connector.name value displays 'WalletConnect'.

We've tried a number of different variations, but have not had success accessing the wallet provider name reliably.

Any thoughts, ideas, links, suggestions, or code snippets to point us in the right direction?

Thanks.

**Comment by glitch-txs** [comment]
When the name is WalletConnect you can get the actual name by using the provider

I think it's`provider.session.peer.name`

**Reply by ghost** [reply]
How do I access provider? Which hook or api is it a part of?

Thanks for the quick feedback.

**Reply by glitch-txs** [reply]
you can use useAccount an get the current connector, from the connector you can call `const provider = await connector.getProvider()`

**Reply by ghost** [reply]
Great.. I'll give it a shot!
